pkg/reconciler/common/components_override.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	mf "github.com/manifestival/manifestival"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"knative.dev/operator/pkg/apis/operator/base"
+)
+
+// ComponentsTransform transforms deployments based on the configuration in `spec.deployment`.
+func ComponentsTransform(obj base.KComponent, log *zap.SugaredLogger) mf.Transformer {
```

Comment:
can't we just pass in the `[]ComponentOverride`, instead of the `KComponent`, so this is usable on other operators that are _not_ implementing the `KComponent` contract?

Reply:
+1